homeassistant/components/tplink/switch.py

```diff
@@ -105,10 +118,14 @@ def is_on(self):
     def turn_on(self, **kwargs):
         """Turn the switch on."""
         self.smartplug.turn_on()
+        self._state = self.smartplug.SWITCH_STATE_ON
+        self.update_state()
 
     def turn_off(self, **kwargs):
         """Turn the switch off."""
         self.smartplug.turn_off()
+        self._state = self.smartplug.SWITCH_STATE_OFF
+        self.update_state()
```

Comment:
Why set the `_state` on the first line, just to overwrite it inside the called function?

Reply:
I was trying to avoid what seemed like a delay in updating the state after turning the device on / off (it appeared to me that is what was already being done in the lights piece so I was also attempting to mimic that.) I don't believe it is needed.

Reply:
I confirmed that this is not needed and am removing the set state. The issue I was seeing was caused by having the sleep before the is_ready check.